sound/soc/soc-pcm.c

```diff
@@ -882,6 +882,9 @@ static int soc_pcm_hw_clean(struct snd_pcm_substream *substream, int rollback)
 			snd_soc_dai_digital_mute(dai, 1, substream->stream);
 	}
 
+	/* run the stream event */
+	snd_soc_dapm_stream_stop(rtd, substream->stream);
+
```

Comment:
Won't this be called twice? The other one is 
```
soc_pcm_close
	-> soc_pcm_clean(substream, 0);
		-> snd_soc_dapm_stream_stop
```

Reply:
@bardliao you are right. This change is in the right directio but needed additional cleanups. I've fixed those now.